internal/adminapi/client_test.go

```diff
@@ -0,0 +1,239 @@
+package adminapi
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var ca *x509.Certificate
+var caPEM *bytes.Buffer
+var caPrivateKeyPEM *bytes.Buffer
+var certPEM *bytes.Buffer
+var certPrivateKeyPEM *bytes.Buffer
+
+func TestMakeHTTPClient(t *testing.T) {
```

Comment:
If I'm not misunderstanding the difference between this test and the one below it is that this one is just testing using the direct, non-file options for providing certificates, so perhaps this naming might be more clear:

```suggestion
func TestMakeHTTPClientWithTLSOpts(t *testing.T) {
```

And then below to help draw the comparison we could call that one `TestMakeHTTPClientWithTLSOptsFromFiles`?

Let me know your thoughts.